packages/core/src/awsService/appBuilder/serverlessLand/wizard.ts

```diff
@@ -22,160 +24,136 @@ export interface CreateServerlessLandWizardForm {
+            placeholder: 'Choose an IaC option for your project',
+            buttons: [vscode.QuickInputButtons.Back],
+        }
+    )
+}
 
-            // Create and show project name input
-            const nameInput = createInputBox({
-                title: 'Enter Project Name',
-                placeholder: 'Enter a name for your new application',
-                buttons: createCommonButtons(),
-                validateInput: (value: string): string | undefined => {
-                    if (!value) {
-                        return 'Application name cannot be empty'
-                    }
-                    if (value.includes(path.sep)) {
-                        return `The path separator (${path.sep}) is not allowed in application names`
-                    }
-                    return undefined
-                },
-            })
+function promptLocation() {
+    return createFolderPrompt(vscode.workspace.workspaceFolders ?? [], {
+        title: localize('AWS.serverlessLand.initWizard.location.prompt', 'Select Project Location'),
+        buttons: [vscode.QuickInputButtons.Back],
+        browseFolderDetail: 'Select a folder for your project',
+    })
+}
 
-            const projectName = await nameInput.prompt()
-            if (!projectName || typeof projectName !== 'string') {
-                return undefined // User cancelled
+function promptName() {
+    return createInputBox({
+        title: localize('AWS.serverlessLand.initWizard.name.prompt', 'Enter Project Name'),
+        placeholder: 'Enter a name for your new application',
+        buttons: [vscode.QuickInputButtons.Back],
+        validateInput: (value: string): string | undefined => {
+            if (!value) {
+                return 'Application name cannot be empty'
             }
-
-            // Return the form with all collected values
-            return {
-                name: projectName,
-                location: selectedLocation,
-                pattern: selectedPattern,
-                runtime: selectedRuntime,
-                iac: selectedIac,
+            if (value.includes(path.sep)) {
+                return `The path separator (${path.sep}) is not allowed in application names`
             }
-        } catch (err) {
-            throw new ToolkitError(`Failed to run wizard: ${err instanceof Error ? err.message : String(err)}`)
-        }
+            return undefined
+        },
+    })
+}
+
+/**
+ * Wizard for creating Serverless Land applications
+ * Guides users through the project creation process
+ */
+export class CreateServerlessLandWizard extends Wizard<CreateServerlessLandWizardForm> {
+    private metadataManager: MetadataManager
+
+    public constructor(context: { defaultRegion?: string; credentials?: AWS.Credentials }) {
+        super({
+            exitPrompterProvider: createExitPrompter,
+        })
+        loadMetadata().catch((err: any) => {
```

Comment:
shouldn't the form have the metadata before the load the form loads rather than loading it afterwords?

Reply:
I don't fully understand what you're talking about here. Can you please explain it more? Thanks!